### PR TITLE
Added configmap and secret checksums to auto restart pods if sha256 changes

### PR DESCRIPTION
**Pull request by smark88**

### CODE REVIEW

**Comment by smark88** [comment]
Renders similar to the following, then any change in the sha will restart the pod. sha changing would mean a secret or configmap val has changed, which to update on the pod requires the pod to restart.
```
      annotations:
        checksum/configmap: f57d28ef6c4e6bd6d3e5854c082528d120b114b7629b0e98d06ce11d4dad2281
```

**Review by snf2ye — APPROVED** [review]

File: charts/datarepo-api/templates/api-deployment.yaml [hunk 1]
```diff
@@ -18,7 +18,12 @@ spec:
         component: {{ include "datarepo-api.fullname" . }}
 ##  dynamically populates environment annotations
       annotations:
+        {{ if .Values.secretsgeneric.datarepoPassword -}}
+        checksum/secrets: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
 {{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+      {{- end }}
```
Comment (hjfbynara) [inline charts/datarepo-api/templates/api-deployment.yaml:26]: I think the {{end}} is in the wrong place.  Unless I am missing something if there is no .values.secretgeneric.datarepPassword - then the .values.podAnnotations will also not be included regardless if they exist.

File: charts/datarepo-ui/templates/ui-deployment.yaml [hunk 2]
```diff
@@ -16,7 +16,12 @@ spec:
         component: {{ include "datarepo-ui.fullname" . }}
 ##  dynamically populates environment annotations
       annotations:
+      {{- if .Values.replicaCount }}
+        checksum/configmap: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
 {{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+      {{- end }}
```
Comment (hjfbynara) [inline charts/datarepo-ui/templates/ui-deployment.yaml:24]: same as above

File: charts/oidc-proxy/templates/oidc-proxy-deployment.yaml [hunk 3]
```diff
@@ -17,7 +17,12 @@ spec:
         component: {{ include "oidc-proxy.fullname" . }}
 ##  dynamically populates environment annotations
       annotations:
+      {{- if .Values.replicaCount }}
+        checksum/configmap: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
 {{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+      {{- end }}
```
Comment (hjfbynara) [inline charts/oidc-proxy/templates/oidc-proxy-deployment.yaml:25]: ditto

**Pull request merged**